deflate.c

```diff
@@ -404,6 +404,8 @@ int32_t Z_EXPORT PREFIX(deflateSetDictionary)(PREFIX3(stream) *strm, const uint8
     while (s->lookahead >= STD_MIN_MATCH) {
         str = s->strstart;
         n = s->lookahead - (STD_MIN_MATCH - 1);
+        for (i = 0; i < n; i++)
+            s->quick_insert_string(s, str + i);
         functable.insert_string(s, str, n);
```

Comment:
Using both functions will overwrite the hashtable entries, effectively nullifying effect of the first two lines.

Reply:
`s->insert_string(s, str, n)` worked, thanks!

Reply:
Only differences between `quick_insert_string` and `insert_string` are that `quick_insert_string` returns the previous entry in the hash table -- so the current string can be compared with another string -- and `insert_string` hashes the remaining bytes of the string, so the latter avoids calling the function inside a loop using indirect jump, when you already know start of the string is a match.